src/App.tsx

```diff
@@ -1,24 +1,131 @@
+  const [filterBy, setFilterBy] = useState(All);
+  const [error, setError] = useState('');
+  const [processing, setProcessing] = useState<number[]>([]);
+
+  const loadTodos = () => {
+    getTodos(USER_ID)
+      .then((todos) => setTodoList(todos))
+      .catch(() => setError(ErrorMessage.Get));
+  };
+
+  useEffect(() => loadTodos(), []);
+
+  useEffect(() => {
+    setTimeout(() => setError(''), 3000);
+  }, [error]);
+
+  const completedTodos = useMemo(() => todoList
+    .filter(todo => todo.completed).length, [todoList]);
+
+  const leftItems = todoList.length - completedTodos;
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  function filterList() {
+    switch (filterBy) {
+      case Active:
+        return todoList.filter((todo: Todo) => !todo.completed);
+
+      case Completed:
+        return todoList
+          .filter((todo: Todo) => !todo.id || todo.completed);
+
+      default:
+        return todoList;
+    }
+  }
+
+  function addTodoToProcesing(id : number | null) {
+    !id ? setProcessing([]) : setProcessing(prev => [...prev, id]);
+  }
+
+  function removeTodo(id: number, sortBy: string) {
+    addTodoToProcesing(id);
+    deleteTodo(id)
+      .then(() => {
+        sortBy === 'completed'
+          ? setTodoList(todoList.filter(todo => !todo.completed))
+          : setTodoList(todoList.filter(todo => todo.id !== id));
+      })
+      .catch(() => setError(ErrorMessage.Delete))
+      .finally(() => addTodoToProcesing(null));
+  }
+
+  function editTodo(id: number, itemToEdit: object, changeTodo: string) {
+    addTodoToProcesing(id);
+    updateTodo(id, itemToEdit)
+      .then(() => setTodoList(todoList.map(currentTodo => {
+        if (changeTodo === 'All') {
+          return { ...currentTodo, ...itemToEdit };
+        }
+
+        return id === currentTodo.id
+          ? { ...currentTodo, ...itemToEdit }
+          : currentTodo;
+      })))
+      .catch(() => setError(ErrorMessage.Patch))
+      .finally(() => addTodoToProcesing(null));
+  }
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          setTodoList={setTodoList}
+          todoList={todoList}
+          setError={setError}
+          addTodoToProcesing={(id:number | null) => addTodoToProcesing(id)}
+          editTodo={
+            (id: number, itemToEdit: object) => editTodo(id, itemToEdit, 'All')
+          }
+        />
+
+        <List
+          setTodoList={setTodoList}
+          todoList={filterList()}
```

Comment:
Don't call the function inside JSX. Move this function to a variable and use memorization for it as well